api/src/main/java/com/xwiki/pdfviewer/macro/PDFViewerMacroParameters.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.pdfviewer.macro;
+
+import org.xwiki.properties.annotation.PropertyAdvanced;
+import org.xwiki.properties.annotation.PropertyDisplayType;
+import org.xwiki.properties.annotation.PropertyMandatory;
+
+import com.xwiki.pdfviewer.PDFResourceReference;
+
+public class PDFViewerMacroParameters
+{
+    /**
+     * The PDF file to be viewer. Use the full attachment reference to specify the PDF file, an absolute URL or only the
+     * name of the attachment when is used along with the document parameter.
+     */
+    private String file;
+
+    /**
+     * The viewer width, in pixels. Uses a percentage value, for example: 25%, 50%, 100%.
+     */
+    private String width = "100%";
+
+    /**
+     * The viewer height, in pixels.
+     */
+    private int height = 1000;
+
+    /**
+     * String reference of the document that contains the file. It is ignored if file is an URL or the resulted
+     * attachment reference does not actually exists.
+     */
+    private String document;
+
+    /**
+     * If this value is true (or 1 or yes) and the user has no access to the  document to which the PDF file is
+     * attached, the PDF file could still be viewed on behalf of the author of the document containing the macro (if
+     * that author has access to the containing document). TODO: change the type to boolean and add a migration for the
+     * old values.
```

Comment:
```suggestion
     * that author has access to the containing document).
     * <p>
     * TODO: Change the type to boolean and add a migration for the old values.
```

I'd make the TODO more visible.